### PR TITLE
Implement "Nest" for `DerivePartialModel` and From `QueryResult`

### PR DESCRIPTION
**Pull request by Goodjooy**
## PR Info

-  #1715

## New Features

- [x] `FromQueryResult` support `flatten`
- [x] `DerivePartialModel` support `flatten`
- [x] document update
- [x] test case

## Changes

- [x] `FromQueryResult` has attributes `sea_orm`

### CODE REVIEW

**Comment by tyt2y3** [comment]
Thank you for the contribution again! You might be a stronger derive macro guru than I am.
Hopefully this will not conflict with the effort "upgrade to syn2" #1713

File: sea-orm-macros/src/derives/from_query_result.rs [hunk 1]
```diff
@@ -20,16 +41,17 @@ pub fn expand_derive_from_query_result(
...
+    let field_query: Vec<TokenStream> = field
         .iter()
-        .map(|f| {
-            let s = f.unraw().to_string();
-            quote! { #s }
+        .map(|FieldInfo { ident, flatten, ty }| {
+            let s = ident.unraw().to_string();
+            if *flatten{
+                quote! { #ident : <#ty as sea_orm::FromQueryResult>::from_query_result(#arg_row, #arg_pre)? }
+            }else{
```
Comment (tyt2y3) [inline sea-orm-macros/src/derives/from_query_result.rs:52]:
```suggestion
            } else {
```

**Comment by Goodjooy** [comment]
> Thank you for the contribution again! You might be a stronger derive macro guru than I am. Hopefully this will not conflict with the effort "upgrade to syn2" #1713

Crate syn with its major version update usually contain one or more **break changes** ，I cannot guarantee there is no  conflict. 

I think I can wait for that PR merged ,then I will handle the conflict in this  PR.

File: sea-orm-macros/src/derives/partial_model.rs [hunk 2]
```diff
@@ -156,6 +166,9 @@ impl DerivePartialModel {
             ColumnAs::Expr { expr, field_name } => {
                 quote!(let #select_ident =  sea_orm::SelectColumns::select_column_as(#select_ident, #expr, #field_name);)
             },
+            ColumnAs::Flatten { ty }=>{
```
Comment (tyt2y3) [inline sea-orm-macros/src/derives/partial_model.rs:169]:
```suggestion
            ColumnAs::Flatten { ty } => {
```

File: sea-orm-macros/src/derives/from_query_result.rs [hunk 3]
```diff
@@ -20,16 +41,17 @@ pub fn expand_derive_from_query_result(
...
         .iter()
-        .map(|f| {
-            let s = f.unraw().to_string();
-            quote! { #s }
+        .map(|FieldInfo { ident, flatten, ty }| {
+            let s = ident.unraw().to_string();
+            if *flatten{
+                quote! { #ident : <#ty as sea_orm::FromQueryResult>::from_query_result(#arg_row, #arg_pre)? }
+            } else {
+                quote! { #ident : #arg_row.try_get(#arg_pre, #s)?}
```
Comment (tyt2y3) [inline sea-orm-macros/src/derives/from_query_result.rs:53]:
```suggestion
                quote! { #ident : #arg_row.try_get(#arg_pre, #s)? }
```

**Comment by tyt2y3** [comment]
Two things:

1) the implementation looks nice!
2) can you add an example or test case showing how the flatten should work? 
3) what does those emoji in front of the commit messages mean? is there a translation table? 👀

**Comment by Goodjooy** [comment]
> Two things:
> 
>     1. the implementation looks nice!
> 
>     2. can you add an example or test case showing how the flatten should work?
> 
>     3. what do those icon in front of the commit messages mean? is there a translation table? 👀

2 things but 3 points :)

The emoji is 'gitmoji' I like using those emoji to mark each commit action.
this is the webside of https://gitmoji.dev/

I'll add examples in the near future.

**Comment by jreppnow** [comment]
(Sorry for coming in from the side..)

Is there a specific reason this was closed? We were looking forward to using this - if required, I can also take a shot at implementing the nesting bit.

**Comment by Goodjooy** [comment]
> (Sorry for coming in from the side..)
> 
> Is there a specific reason this was closed? We were looking forward to using this - if required, I can also take a shot at implementing the nesting bit.

Why this pr closed? I had never close it?

**Comment by Goodjooy** [comment]
However, the code is a bit old, I need time to read and refactor it

**Comment by jreppnow** [comment]
@Goodjooy Hi again、I am really sorry for coming in from the side, but since we kind of desperately need this feature, I decided to do a separate implementation in #2179 . I honestly only care about this feature landing as quickly as possible, so I would not mind if you take my code and update your PR, if you prefer that path instead of my PR superseding yours.

**Comment by Goodjooy** [comment]
> @Goodjooy Hi again、I am really sorry for coming in from the side, but since we kind of desperately need this feature, I decided to do a separate implementation in #2179 . I honestly only care about this feature landing as quickly as possible, so I would not mind if you take my code and update your PR, if you prefer that path instead of my PR superseding yours.
> 
> CC: @tyt2y3

Maintaining  your own modified version of the nest feature might be a more ideal way right now. Adding features usually needs a minor version update which might waiting a bit longer than bug fixing. Not to mention that the core maintainers now are busy in releasing the 1.0 version right now.

**Comment by tyt2y3** [comment]
Hey, thank you for everyone's involvement. I think the only things missing from this PR are some tests and a bit of doc example?

**Comment by Goodjooy** [comment]
there was a better one #2179 that this pr can be close

**Pull request closed without merging**